cli/src/commands/simplify_parents.rs

```diff
@@ -57,63 +54,53 @@ pub(crate) fn cmd_simplify_parents(
     let num_orig_commits = commit_ids.len();
 
     let mut tx = workspace_command.start_transaction();
-    let mut stats = SimplifyStats::default();
+    let mut simplified_commits = 0;
+    let mut edges = 0;
+    let mut reparented_descendants = 0;
+
     tx.repo_mut()
-        .transform_descendants(command.settings(), commit_ids, |rewriter| {
-            if commit_ids_set.contains(rewriter.old_commit().id()) {
-                simplify_commit_parents(command.settings(), rewriter, &mut stats)?;
+        .transform_descendants(command.settings(), commit_ids, |mut rewriter| {
+            if !commit_ids_set.contains(rewriter.old_commit().id())
+                || rewriter.old_commit().parent_ids().len() <= 1
+            {
+                if rewriter.parents_changed() {
+                    rewriter.reparent(command.settings())?.write()?;
+                    reparented_descendants += 1;
+                }
+                return Ok(());
             }
 
+            let num_old_heads = rewriter.new_parents().len();
+            rewriter.simplify_ancestor_merge();
```

Comment:
nit: may be the `if ..` condition can be inserted here? We'll do `reparent()` anyway if ancestors were simplified.

Reply:
The intent of putting the `if` block at the start was to avoid calling `rewriter.simplify_ancestor_merge()` when unneeded.

Reply:
Yes. I just meant
```rust
let num_old_heads = ..;
if target_rev && num_old_heads > 1 {
    rewriter.simplify_ancestor_merge();
}
```